website/docs/migration.md

```diff
@@ -0,0 +1,244 @@
+---
+sidebar_position: 3
+---
+
+# Migrating from MongoDB to FerretDB
+
+To ensure a smooth and successful migration from MongoDB, we offer several methods through which you can test your application with FerretDB.
+
+## Operation modes
+
+We support different operation modes, by default FerretDB will always run on `normal` mode.
+In this mode all client requests are processed _only_ by FerretDB and returned to the client.
+
+You can set modes using either the `--mode` flag or the `FERRETDB_MODE` environment variable, accepting values such as `normal`, `proxy`, `diff-normal`, and `diff-proxy`.
+
+### Manual and automated testing with `diff-normal` mode
+
+`diff-normal` mode acts a proxy and will forward client requests to another MongoDB-compatible database, and will log the difference between them.
+
+You can manually test your application or use integration tests, among other methods.
+Afterward, you can inspect the differential output for errors or inconsistencies between responses that require your attention.
+
+As an example, let us say that your application performs some complex query and you'd like to test it in `diff-normal` mode.
+You would do the following:
+
+1. Start the environment and run FerretDB.
+
+   ```bash
+   # in a terminal run env-up to start the environment
+   me@foobar:~/FerretDB$ bin/task env-up
```

Comment:
This documentation is for users, not for developers. Users don't have the source code of FerretDB or `bin/task`. We should link to https://docs.ferretdb.io/quickstart-guide/ and https://docs.ferretdb.io/configuration/flags/ and then explain how to use those flags to enable proxy mode

Reply:
@Fashander I think the first review should focus on big things like that ^